File: libraries/python/examples/example_single.py
"""Example: running one simulation with manual progress reporting."""

from pathlib import Path

from trnrun import SimulationConfig, SimulationManager

__root__ = Path(__file__).resolve().parent

config = SimulationConfig(watch_tmp=True)

with SimulationManager(max_concurrent=1, refresh_interval=0.1) as manager:
    sim = manager.add(__root__ / "dck" / "example_wo_plot_w_tracking.dck", config)
    manager.wait(sim)

File: libraries/matlab/toolbox/examples/example_single.m
function simulation = example_single()
%EXAMPLE_SINGLE Run one simulation with manual progress reporting.
%   SIMULATION = EXAMPLE_SINGLE() submits the bundled Type3830 example deck
%   through a manager limited to one concurrent run and waits for it to
%   finish. Function scope keeps the onCleanup guard deterministic.

    example_folder = string(fileparts(mfilename("fullpath")));
    addpath(fileparts(example_folder))
    import trnrun.*

    deck_file = fullfile(example_folder, "dck", "example_wo_plot_w_tracking.dck");

    config = SimulationConfig(watch_tmp=true);
    manager = SimulationManager(maxConcurrent=1, refreshInterval=0.1);

    simulation = manager.add(deck_file, config);
    manager.wait(simulation);
    manager.shutdown();
end
